data/nat/choose.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+Mostly based on Jeremy Avigad's choose file in lean 2
+-/
+
+
+import data.nat.basic
+
+open nat
+def choose : ℕ → ℕ → ℕ
+| _              0  := 1
+| 0        (succ k) := 0
+| (succ n) (succ k) := choose n k + choose n (succ k)
```

Comment:
The equation compiler and the elaborator understand that `succ n = n + 1`, so it's nicer to use `n + 1` instead of `succ`.